bothan-coingecko/src/api/rest.rs

```diff
@@ -21,57 +20,54 @@ impl CoinGeckoRestAPI {
     pub async fn get_coins_list(&self) -> Result<Vec<Coin>, RestAPIError> {
         let url = format!("{}coins/list", self.url);
         let builder = self.client.get(url);
-        let response = send_request(builder).await?;
 
-        Ok(response.json::<Vec<Coin>>().await?)
+        request::<Vec<Coin>>(builder).await
     }
 
     /// Retrieves market data for the specified coins from the CoinGecko API.
-    pub async fn get_coins_market(
+    pub async fn get_coins_market<T: AsRef<str>>(
         &self,
-        ids: &[&str],
-        page_size: usize,
-        page: usize,
-    ) -> Result<Vec<Option<Market>>, RestAPIError> {
+        ids: &[T],
+        order: Option<Order>,
```

Comment:
I think the caller should always be providing these options. Having default value means the caller may not get prices of all the supplied coins.

Reply:
I've changed it to omit the parameter if None is given to use the endpoints default value as the endpoint doesn't require them.